### PR TITLE
Set default_flow_style=None in yaml.dump calls

### PR DESCRIPTION
**Pull request by Ch3LL**
### What does this PR do?

In PyYaml 5.1 they changed the `default_flow_style` from None to False. Changing this back to None in salt since this is a breaking change to keep backwards compatibility. Here is the PR/commit that added this: https://github.com/yaml/pyyaml/pull/256 https://github.com/yaml/pyyaml/commit/507a464ce62c933bf667b2296a96ad45f0147873

Would appreciate any further review as I'm not sure if we want to keep backwards compatibility or update salt to ensure `default_flow_style=False` works properly. But I think this is the least obtrusive change for now.

### What issues does this PR fix or reference?
Fixes up some tests on Fedora which has PyYaml 5.1 installed:

https://github.com/saltstack/salt/issues/52930
https://github.com/saltstack/salt/issues/52929
https://github.com/saltstack/salt/issues/52928
https://github.com/saltstack/salt/issues/52927

PyYAML 5.1:

```
>>> test = {'foo': 'bar'}
>>> yaml.dump(test)
'foo: bar\n'
```

PyYAML 3.13:

```
>>> test = {'food': 'bar'}
>>> yaml.dump(test)
'{food: bar}\n'
```

With this change it will always dump like so:

PyYAML 5.1:

```
>>> yaml.dump(test, default_flow_style=None)
'{food: bar}\n'
```

PyYAML 3.13:

```
>>> yaml.dump(test, default_flow_style=None)
'{food: bar}\n'
```

### Tests written?

Yes

### Commits signed with GPG?

Yes

### CODE REVIEW

File: tests/unit/utils/test_yamldumper.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+# Import python libs
+from __future__ import absolute_import, print_function, unicode_literals
+
+# Import Salt Libs
+import salt.utils.yamldumper
+
+# Import Salt Testing Libs
+from tests.support.unit import TestCase, skipIf
+from tests.support.mock import patch, NO_MOCK, NO_MOCK_REASON, mock_open
```
Comment (waynew) [inline tests/unit/utils/test_yamldumper.py]:
```suggestion
from tests.support.mock import NO_MOCK, NO_MOCK_REASON
```

According to pylint 😛

**Review by waynew — CHANGES_REQUESTED** [review]

File: tests/unit/utils/test_yamldumper.py (hunk 1 above)
Comment (Ch3LL) [reply tests/unit/utils/test_yamldumper.py]: bah! thanks for that :)

**Comment by Ch3LL** [comment]
started a full test run:

https://jenkinsci.saltstack.com/job/2019.2.1/job/ch3ll-yaml-salt-fedora-29-py2/
https://jenkinsci.saltstack.com/job/2019.2.1/job/ch3ll-yaml-salt-fedora-29-py3/

**Review by waynew — APPROVED** [review]

File: tests/unit/utils/test_yamldumper.py (hunk 1 above)
Comment (waynew) [reply tests/unit/utils/test_yamldumper.py]: BTDT. Too many times 🙃

**Comment by waynew** [comment]
I bet there's a unicode issue on py3. When we're using pytest this assertion style will actually give us information 😄 Of course, right now... 😢 

You're probably aware of both these options, but you could do
`self.assertEqual(salt.utils.yamldumper.dump(data), '{!!python/unicode \'foo\': !!python/unicode \'bar\'}\n')` for now. Or you could do

```
actual = salt.utils.yamldumper.dump(data)
assert actual == '{!!python/unicode \'foo\': !!python/unicode \'bar\'}\n', actual
``` to see what the actual text is.

**Comment by Ch3LL** [comment]
i'm fine with just sticking to just `assert`. I know we do not get more information, but this will make it an easier transition to pytest and will get more information then.

**Review by garethgreenaway — APPROVED** [review]

**Pull request merged**